src/core/ledger/currency.js

```diff
@@ -82,3 +82,19 @@ export const currencyIdParser: C.Parser<CurrencyId> = C.orElse([
   evmIdParser,
   protocolIdParser,
 ]);
+
+export function buildCurrencyId(
+  chainId: string,
+  currencyAddress?: string
+): CurrencyId {
```

Comment:
Calling this CurrencyId makes me thing it behaves like a string since almost all instances of "IDs" that I can think of have been strings